src/plugins/kernel_install/utils.sh

```diff
@@ -289,12 +289,28 @@ function list_installed_kernels()
   local prefix="$4"
   local -a available_kernels=()
   local cmd
+  local file_system_type
+  local ret=0
 
   # TODO: Drop me in the future
   migrate_old_kernel_list
 
-  cmd_manager "$flag" "sudo mkdir -p $REMOTE_KW_DEPLOY"
-  cmd_manager "$flag" "sudo touch $INSTALLED_KERNELS_PATH"
+  file_system_type=$(detect_filesystem_type '')
+  is_filesystem_writable "$file_system_type"
```

Comment:
is_filesystem_writable "$file_system_type" "$flag"

Otherwise it will print the "is_filesystem_writable" command everytime.